plugins/arangodbendorsements/arangodb.go

```diff
@@ -0,0 +1,204 @@
+	dbparams      ArangoDBparams
+	connvars      ArangoDBConnVars
+}
+
+// NewArangoStore instantiates a new Arango store
+func NewArangoStore(dbparams ArangoDBparams) (*ArangoStore, error) {
+	as := &ArangoStore{}
+	if err := as.init(dbparams); err != nil {
+		return nil, err
+	}
+	return as, nil
+}
+
+func (e *ArangoStore) init(dbparams ArangoDBparams) error {
+	e.dbparams = dbparams
+	e.isInitialised = true
+	return nil
+}
+
+// Connect is responsible for making connection to the arangoDB
+func (e *ArangoStore) Connect(ctx context.Context) error {
+	var ok bool
+	var err error
+
+	// Create an HTTP Connection First to the Client
+	e.connvars.conn, err = http.NewConnection(http.ConnectionConfig{
+		Endpoints: []string{"http://localhost:8529"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create HTTP connection: %w", err)
+	}
+
+	// Create a client
+	e.connvars.client, err = driver.NewClient(driver.ClientConfig{
+		Connection:     e.connvars.conn,
+		Authentication: driver.BasicAuthentication(e.dbparams.Login, e.dbparams.Password),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create a client: %w", err)
+	}
+
+	// Check if Database with a given name exists or not
+	ok, err = e.connvars.client.DatabaseExists(ctx, e.dbparams.StoreName)
+	if err != nil {
+		return fmt.Errorf("failure while checking whether DB exists: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("endorsement database %s does not exist", e.dbparams.StoreName)
+	}
+	e.connvars.db, err = e.connvars.client.Database(ctx, e.dbparams.StoreName)
+	if err != nil {
+		return fmt.Errorf("failed to connect to the Endorsement database: %w", err)
+	}
+
+	// Check if the Graph exists in the DB
+	ok, err = e.connvars.db.GraphExists(ctx, e.dbparams.GraphName)
+	if err != nil {
+		return fmt.Errorf("failure while checking whether graph exists: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("endorsement graph %s does not exist", e.dbparams.GraphName)
+	}
+	// Connect to the Graph
+	e.connvars.graph, err = e.connvars.db.Graph(ctx, e.dbparams.GraphName)
+	if err != nil {
+		return fmt.Errorf("failed to connect to %s: %w", e.dbparams.GraphName, err)
+	}
+
+	log.Printf("connection to %s DB and %s graph succeeded!", e.dbparams.StoreName, e.dbparams.GraphName)
+	e.connvars.paramValid = true
+	return nil
+}
+
+// IsInitialised returns whether store is initialized or not?
+func (e *ArangoStore) IsInitialised() error {
+	if e.isInitialised == false {
+		return fmt.Errorf("arango store is uninitialised")
+	}
+	return nil
+}
+
+// GetQueryParam returns a suitable query parameter based on supplied input
+func (e *ArangoStore) GetQueryParam(input interface{}) (string, error) {
+	switch input.(CollectionType) {
+	case HW:
+		return e.dbparams.HwCollection, nil
+	case SW:
+		return e.dbparams.SwCollection, nil
+	case Edge:
+		return e.dbparams.EdgeCollection, nil
+	case Rel:
+		return e.dbparams.RelCollection, nil
+	default:
+		return "", fmt.Errorf("invalid input to GetQueryParam")
```

Comment:
if you are changing the input to a string, add which "invalid input" has been supplied